misk/src/test/kotlin/misk/tokens/FakeTokenGeneratorTest.kt

```diff
@@ -51,6 +52,14 @@ class FakeTokenGeneratorTest {
     assertThat(token3).isEqualTo("payment000000000000000002")
   }
 
+  @Test
+  fun generationIsThreadSafe() {
```

Comment:
This isn’t testing what you think it’s testing. Because you aren’t passing 100 unique prefixes to tokenGenerator.generate() you’re making it extremely unlikely that unsynchronized accesses to the map will contend in a meaningful way.

I think you want `tokenGenerator.generate("token$it")`

Reply:
I red/green tested this and it exposes the bug (revert to `mutableMapOf` to see it). The bug being contention around first access for a single key.

Reply:
Got it. Works for me.